src/fmm.jl

```diff
@@ -0,0 +1,296 @@
+    end
+
+    @views corblkasm = BEAST.blockassembler(
+        operator,
+        test_functions,
+        trial_functions,
+        quadstrat=quadstratcbk
+    )
+
+    @views function corassembler(Z, tdata, sdata)
+        @views store(v,m,n) = (Z[m,n] += v)
+        corblkasm(tdata, sdata, store)
+    end
+
+    MBF = FastBEAST.MatrixBlock{Int, scalartype(operator), Matrix{scalartype(operator)}}
+    fullrankblocks = MBF[]
+    fullrankblocks_perthread = Vector{MBF}[]
+    correctionblocks = MBF[]
+    correctionblocks_perthread = Vector{MBF}[]
+
+    test_tree = create_tree(test_functions.pos, FastBEAST.BoxTreeOptions(nmin=nmin))
+    trial_tree = create_tree(trial_functions.pos, FastBEAST.BoxTreeOptions(nmin=nmin))
+
+    fullinteractions = []
+    compressableinteractions = []
+
+    FastBEAST.computerinteractions!(
+        test_tree,
+        trial_tree,
+        fullinteractions,
+        compressableinteractions
+    )
+
+    if threading == :single
+        for fullinteraction in fullinteractions
+            push!(
+                fullrankblocks,
+                FastBEAST.getfullmatrixview(
+                    assembler,
+                    fullinteraction[1],
+                    fullinteraction[2],
+                    Int,
+                    scalartype(operator)
+                )
+            )
+            push!(
+                correctionblocks,
+                FastBEAST.getfullmatrixview(
+                    corassembler,
+                    fullinteraction[1],
+                    fullinteraction[2],
+                    Int,
+                    scalartype(operator)
+                )
+            )
+        end
+    elseif threading == :multi
+        for i in 1:Threads.nthreads()
+            push!(fullrankblocks_perthread, MBF[])
+            push!(correctionblocks_perthread, MBF[])
+        end
+
+        Threads.@threads for fullinteraction in fullinteractions
+            push!(
+                fullrankblocks_perthread[Threads.threadid()],
+                FastBEAST.getfullmatrixview(
+                    assembler,
+                    fullinteraction[1],
+                    fullinteraction[2],
+                    Int,
+                    scalartype(operator)
+                )
+            )
+            push!(
+                correctionblocks_perthread[Threads.threadid()],
+                FastBEAST.getfullmatrixview(
+                    corassembler,
+                    fullinteraction[1],
+                    fullinteraction[2],
+                    Int,
+                    scalartype(operator)
+                )
+            )
+
+        end
+
+        for i in eachindex(fullrankblocks_perthread)
+            append!(fullrankblocks, fullrankblocks_perthread[i])
+        end
+        for i in eachindex(correctionblocks_perthread)
+            append!(correctionblocks, correctionblocks_perthread[i])
+        end
+    end
+
+    return fullrankblocks, correctionblocks, fullinteractions
+end
+
```

Comment:
I would explain in a little bit more detail, what is going on here. I am not sure a mesh can be mapped to points.